src/hooks/useQuery.tsx

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react'
+import type { Response } from '../types'
+
+export const useQuery = (query: any) => {
```

Comment:
anyway to make this accept a generic and pass it to the state? would be useful for DX